### PR TITLE
Test deep_dta

### PR DESCRIPTION
**Pull request by pz-white**
### Description
1. Add tests covering deepdta related modules including:
- kale/embed/seq_nn.py
- kale/loaddata/tdc_datasets.py
- kale/pipeline/deep_dti.py
- kale/predict/decode.py
- kale/prepdata/chem_transform.py

2. Introduce conda virtual environment in test action.

3. Remove MNIST/MNISTM test as they are blocked due to HTTPServer error.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.

### CODE REVIEW

File: pytest.ini
Comment (haipinglu) [inline pytest.ini:1]: This is a good direction to resolve the challenge but we prefer to keep the number of files small so this should be configured in `pyproject.toml` instead. See [pytest docs on config using `pyproject.toml`](https://docs.pytest.org/en/stable/customize.html#pyproject-toml).

**Comment by pz-white** [comment]
Reproduce the url error for MINIST, try to re-run after the server is stable.

**Comment by haipinglu** [comment]
> Reproduce the url error for MINIST, try to re-run after the server is stable.

That's annoying. Hope it is temporary, otherwise, we will consider to resolve or skip it (MNIST/M) later.

**Comment by bobturneruk** [comment]
Introduces `rdkit` dependency not installed by `pip install -e .[dev]`.

**Comment by bobturneruk** [comment]
@pz-white can you run these tests locally in a clean `conda` environment using `pytest`?

**Comment by pz-white** [comment]
@bobturneruk yes I can successfully run these test with pytest locally with my conda environment

File: .github/workflows/test.yml
Comment (bobturneruk) [inline .github/workflows/test.yml:31]: If we need conda, consider https://github.com/conda-incubator/setup-miniconda

**Comment by bobturneruk** [comment]
Seems like this PR introduces a lot of new requirements and changes CI quite a bit - we need to decide on how these will be managed. My default suggestion is to add to `setup.py` where possible, although I appreciate that `rdkit` needs `conda`. I suppose this means that `PyKale` now also requires `conda` unless the required functionality in `rdkit` can be installed some other way?

File: .github/workflows/test.yml
Comment (pz-white) [reply .github/workflows/test.yml:31]: Thanks Bob! Let me try it now.

**Comment by haipinglu** [comment]
> Seems like this PR introduces a lot of new requirements and changes CI quite a bit - we need to decide on how these will be managed. My default suggestion is to add to `setup.py` where possible, although I appreciate that `rdkit` needs `conda`. I suppose this means that `PyKale` now also requires `conda` unless the required functionality in `rdkit` can be installed some other way?

Not sure how to do it. The official instruction to install `pytorch geometric` is bit different from regular packages, as in the updated `test.yml`. Do you think this can be put in `setup.py`? If yes, can you make the change? I don't know how yet.

File: .github/workflows/test.yml
Comment (pz-white) [reply .github/workflows/test.yml:31]: The conda should be necessary as I checked that the `rdkit` is not a pure python package so that the package author said it is not supported to distribute with `PyPi`

**Comment by bobturneruk** [comment]
I'll have a look. I think this may impact on `PyKale` install instructions. I hope it doesn't mean extra manual steps.

**Comment by bobturneruk** [comment]
I think we need to make it clearer what needs to be installed to run the tests: https://pykale.readthedocs.io/en/latest/installation.html

**Comment by haipinglu** [comment]
> I think we need to make it clearer what needs to be installed to run the tests: https://pykale.readthedocs.io/en/latest/installation.html

Our current suggestion is pytorch and pytorch geometric should be manually installed, which is the most common practice because this often needs to match the hardware (cpu/gpu versions). 

Only those who works on graphs and use related APIs need to install pytorch geometric, not everybody (among 6 of us, only Peizhen is working on graphs. ). We have MSc students followed the instructions and it seems all right.

File: tests/embed/test_seq_nn.py
Comment (haipinglu) [inline tests/embed/test_seq_nn.py:38]: A bit hard to understand for me here, to learn from you later, maybe just because I am not familiar. If complicated, add some light comments may help others.

**Review by haipinglu — APPROVED** [review]
Good job!
@bobturneruk Shall we wait for your review before merging? Please let us know. Thanks.

**Comment by bobturneruk** [comment]
Please don't wait. If there's anything specific you want me to look at retrospectively, let me know.

File: .github/workflows/test.yml
Comment (haipinglu) [inline .github/workflows/test.yml:55]: @pz-white May I know why you changed `bash` to `bash -l {0}`? Now [codecov](https://codecov.io/gh/pykale/pykale/tree/master/kale) is not updating and I am suspecting this to be the cause.

**Pull request merged**
